magento/magento2
# Generate critical css file for blank theme

### Description (*)

Generate critical css following below guide for blank theme
https://devdocs.magento.com/guides/v2.4/frontend-dev-guide/css-topics/css-critical-path.html
Default magento only have critical css for luma theme

Critical css is one of factor improvement page speed render. For browser render content above the fold as soon as possible


> How to generate css critical
We have some ways for generate critical 
One choice we can use this repo
But required enviroment setup NodeJS/Npm/Gulp

https://github.com/bobmotor/magento-2-gulp 
This repo current version up to 1.5.0 - used gulp version 4
=> File critical.css output ~26kB

Step following https://github.com/bobmotor/magento-2-gulp/blob/master/README.md
At the last step we use lighthouse for render output critical.css for particular url page (/home, category, details)
We also need some tweak change (clear some declare with font, exclude styles related with email) , lastly modify critical styles following rule
-File size should not too big for block render page (Idealy ~20-50Kb)
-Styles for critical should not overwrite styles declared later (styles-m, styles-l)


### Related Pull Requests


### Fixed Issues (if relevant)

1. Fixes magento/magento2#<issue_number>

No issue available

### Manual testing scenarios (*)

1. Enable critical path in backend. Or setup via cli command
bin/magento config:set dev/css/use_css_critical_path 1
2. Make sure set blank theme as default in admin 
3. Clear full_page and config cache
4. Go to homepage view page source
5. See critical loaded in head tag of DOM
6. Reload various pages (home, category, details product)  without broken styles visual

### Questions or comments

CC: @vzabaznov or @adifucan Feel free to give me any advices
@Leland can you help me review this ? Thanks

### Contribution checklist (*)


### Resolved issues:
1. [x] resolves magento/magento2#31527: Generate critical css file for blank theme
